### ISSUE TITLE
Failure to install on CiviCRM 4.7.22/Wordpress 4.8

### ISSUE THREAD

**Issue by lathem** [JMAConsulting/biz.jmaconsulting.grantapplications#132]
When attempting to install the Grant Applications extension, I get the following error:

CRM_Exception: Could not find the SQL file. in /var/www/dearproject.org/wp-content/uploads/civicrm/ext/biz.jmaconsulting.grantapplications/CRM/Grantapplications/Upgrader/Base.php on line 259

I checked, and the sql files are indeed present, so I'm not sure where to go from here. 

My system is a brand new install of:
PHP 7.0.19
MySQL 5.5
Apache 2.4.25
Wordpress 4.8
CiviCRM 4.7.22

Also installed are the extensions:
CiviMobile 2.0
Angular Profiles 4.6-1.0.2
CiviVolunteer 4.6-2.2.1

**Comment by Edzelopez** [comment]
Hi @lathem 

The extension currently only supports installations upto 4.7.14. However, for a quick fix, replacing
CRM_Utils_File::sourceSQLFile(CIVICRM_DSN, $smarty->fetch($file), NULL, TRUE);

With 

CRM_Utils_File::sourceSQLFile(CIVICRM_DSN, $file);

should get the installation SQL picked up by CiviCRM. 

However, I presume there might be other SQL errors while installing since CiviCRM has changed a lot of methods around extensions so you might be better off executing each line of code separately from the grantapplications_install.sql file to your database. You can then comment out the lines from the SQL file and allow the extension to install as usual.

**Comment by lathem** [comment]
Thanks for your super fast reply, @Edzelopez 
Unfortunately, it kind of raised more questions than it answered.  I made the change at line 259 of base.php  as you suggested, as just like you predicted, it fails because of SQL errors. 

So, as you said, I executed the SQL commands one at a time, and I've pasted in the output I got.  I'm not a good enough SQL whiz to spot what's happening, though.   Since I don't have any actual customer data loaded into this install, would I be better off just going to version 4.7.14? 


`MariaDB [(none)]> use civicrm

Database changed
MariaDB [civicrm]> CREATE TABLE IF NOT EXISTS `civicrm_grant_app_page` (
    ->   `id` int(10) unsigned NOT NULL AUTO_INCREMENT COMMENT 'Grant Application Page Id.',
    ->   `title` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL COMMENT 'Grant Application Page title. For top of page display.',
    ->   `intro_text` text COLLATE utf8_unicode_ci COMMENT 'Text and html allowed. Displayed below title.',
    ->   `footer_text` text COLLATE utf8_unicode_ci COMMENT 'Text and html allowed. Displayed at the bottom of the first page of the grant application wizard.',
    ->   `grant_type_id` int(10) unsigned NOT NULL COMMENT 'Grant type assigned to applications submitted via this page.',
    ->   `default_amount` decimal(20,2) DEFAULT NULL COMMENT 'Default amount of grant applied for.',
    ->   `is_draft` tinyint(4) DEFAULT NULL COMMENT 'Does this page have a Save as Draft button?',
    ->   `draft_title` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL COMMENT 'Title for Save as Draft page (header title tag, and display at the top of the page).',
    ->   `draft_text` text COLLATE utf8_unicode_ci COMMENT 'Text and html allowed. Displayed above application fields on Save as Draft page',
    ->   `draft_footer` text COLLATE utf8_unicode_ci COMMENT 'Text and html allowed. Displayed at the bottom of the Save as Draft page.',
    ->   `thankyou_title` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL COMMENT 'Title for Thank-you page (header title tag, and display at the top of the page).',
    ->   `thankyou_text` text COLLATE utf8_unicode_ci COMMENT 'Text and html allowed. Displayed above result on success page',
    ->   `thankyou_footer` text COLLATE utf8_unicode_ci COMMENT 'Text and html allowed. displayed at the bottom of the success page. Common usage is to include link(s) to other pages such as tell-a-friend, etc.',
    ->   `confirm_text` text COLLATE utf8_unicode_ci COMMENT 'Text and html allowed. Displayed above result on confirmation page',
    ->   `confirm_footer` text COLLATE utf8_unicode_ci COMMENT 'Text and html allowed. displayed at the bottom of the confirmation page.',
    ->   `is_email_receipt` tinyint(4) DEFAULT '0' COMMENT 'If true, receipt is automatically emailed to contact on success',
    ->   `receipt_from_name` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL COMMENT 'FROM email name used for receipts generated by applications to this grant application page.',
    ->   `receipt_from_email` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL COMMENT 'FROM email address used for receipts generated by applications to this grant application page.',
    ->   `cc_receipt` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL COMMENT 'Comma-separated list of email addresses to cc each time a receipt is sent',
    ->   `bcc_receipt` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL COMMENT 'Comma-separated list of email addresses to bcc each time a receipt is sent',
    ->   `receipt_text` text COLLATE utf8_unicode_ci COMMENT 'Text to include above standard receipt info on receipt email. emails are text-only, so do not allow html for now',
    ->   `is_active` tinyint(4) DEFAULT NULL COMMENT 'Is this grant application page active?',
    ->   `start_date` datetime DEFAULT NULL COMMENT 'Date and time that this page starts.',
    ->   `end_date` datetime DEFAULT NULL COMMENT 'Date and time that this page ends. May be NULL if no defined end date/time',
    ->   `created_id` int(10) unsigned DEFAULT NULL COMMENT 'FK to civicrm_contact, who created this grant application page',
    ->   `created_date` datetime DEFAULT NULL COMMENT 'Date and time that grant application page was created.',
    ->   `is_for_organization` tinyint(4) DEFAULT '0' COMMENT 'if true, signup is done on behalf of an organization',
    ->   `for_organization` text COLLATE utf8_unicode_ci COMMENT 'This text field is shown when is_for_organization is checked. For example - I am submitting grant application on behalf of an organization.',
    ->   PRIMARY KEY (`id`),
    ->   KEY `FK_civicrm_grant_app_page_created_id` (`created_id`)
    -> ) ENGINE=InnoDB  DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
Query OK, 0 rows affected, 1 warning (0.00 sec)

MariaDB [civicrm]> ALTER TABLE `civicrm_grant_app_page`
    ->   ADD CONSTRAINT `FK_civicrm_grant_app_page_created_id` FOREIGN KEY (`created_id`) REFERENCES `civicrm_contact` (`id`) ON DELETE SET NULL;
ERROR 1005 (HY000): Can't create table `civicrm`.`#sql-267_15ed` (errno: 121 "Duplicate key on write or update")
MariaDB [civicrm]> SELECT @parentId := id FROM `civicrm_navigation` WHERE `name` = 'Grants';
+-----------------+
| @parentId := id |
+-----------------+
|             105 |
+-----------------+
1 row in set (0.01 sec)

MariaDB [civicrm]> INSERT INTO `civicrm_navigation` (`domain_id`, `label`, `name`, `url`, `permission`, `permission_operator`, `parent_id`, `is_active`, `has_separator`, `weight`) VALUES
    -> ({$domainID}, 'New Grant Application Page', 'New Grant Application Page', 'civicrm/admin/grant/apply?reset=1&action=add', 'access CiviGrant', 'AND', @parentId, 1, 1, 4);
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '}, 'New Grant Application Page', 'New Grant Application Page', 'civicrm/admin/gr' at line 2
MariaDB [civicrm]> INSERT INTO `civicrm_navigation` (`domain_id`, `label`, `name`, `url`, `permission`, `permission_operator`, `parent_id`, `is_active`, `has_separator`, `weight`) VALUES
    -> ({$domainID}, 'New Grant Application Page', 'New Grant Application Page', 'civicrm/admin/grant/apply?reset=1&action=add', 'access CiviGrant', 'AND', @parentId, 1, 1, 4);
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '}, 'New Grant Application Page', 'New Grant Application Page', 'civicrm/admin/gr' at line 2
MariaDB [civicrm]> SELECT @optionGroupId := id FROM `civicrm_option_group` WHERE `name` = 'activity_type';
+----------------------+
| @optionGroupId := id |
+----------------------+
|                    2 |
+----------------------+
1 row in set (0.00 sec)

MariaDB [civicrm]> SELECT @maxValue := MAX( CAST( `value` AS UNSIGNED ) ) + 1 FROM  `civicrm_option_value` WHERE `option_group_id` = @optionGroupId;
+-----------------------------------------------------+
| @maxValue := MAX( CAST( `value` AS UNSIGNED ) ) + 1 |
+-----------------------------------------------------+
|                                                  57 |
+-----------------------------------------------------+
1 row in set (0.00 sec)

MariaDB [civicrm]>
MariaDB [civicrm]> SELECT @maxWeight := MAX( CAST( `weight` AS UNSIGNED ) ) + 1 FROM  `civicrm_option_value` WHERE `option_group_id` = @optionGroupId;
+-------------------------------------------------------+
| @maxWeight := MAX( CAST( `weight` AS UNSIGNED ) ) + 1 |
+-------------------------------------------------------+
|                                                    57 |
+-------------------------------------------------------+
1 row in set (0.00 sec)

MariaDB [civicrm]>
MariaDB [civicrm]> SELECT @activityTypeId := id FROM `civicrm_option_value` WHERE `name` = 'Grant';
Empty set (0.00 sec)

MariaDB [civicrm]>
MariaDB [civicrm]> INSERT IGNORE INTO `civicrm_option_value` (`id`, `option_group_id`, {localize field='label'}`label`{/localize}, `value`, `name`, `grouping`, `filter`, `is_default`, `weight`, {localize field='description'}`description`{/localize}, `is_optgroup`, `is_reserved`, `is_active`, `component_id`, `domain_id`, `visibility_id`) VALUES
    -> (@activityTypeId, @optionGroupId, {localize}'{ts escape="sql"}Grant{/ts}'{/localize}, @maxValue, 'Grant', NULL, 1, NULL, @maxWeight, {localize}'Online Grant Application'{/localize}, 0, 1, 1, 5, NULL, NULL);
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '{localize field='label'}`label`{/localize}, `value`, `name`, `grouping`, `filter' at line 1
MariaDB [civicrm]> SELECT @dashId := id FROM `civicrm_option_group` WHERE `name` = 'user_dashboard_options';
+---------------+
| @dashId := id |
+---------------+
|            20 |
+---------------+
1 row in set (0.00 sec)

MariaDB [civicrm]>
MariaDB [civicrm]> SELECT @maxValue := MAX( CAST( `value` AS UNSIGNED ) ) + 1 FROM  `civicrm_option_value` WHERE `option_group_id` = @dashId;
+-----------------------------------------------------+
| @maxValue := MAX( CAST( `value` AS UNSIGNED ) ) + 1 |
+-----------------------------------------------------+
|                                                  11 |
+-----------------------------------------------------+
1 row in set (0.00 sec)

MariaDB [civicrm]>
MariaDB [civicrm]> SELECT @maxWeight := MAX( CAST( `weight` AS UNSIGNED ) ) + 1 FROM  `civicrm_option_value` WHERE `option_group_id` = @dashId;
+-------------------------------------------------------+
| @maxWeight := MAX( CAST( `weight` AS UNSIGNED ) ) + 1 |
+-------------------------------------------------------+
|                                                    11 |
+-------------------------------------------------------+
1 row in set (0.00 sec)

MariaDB [civicrm]>
MariaDB [civicrm]> INSERT IGNORE INTO `civicrm_option_value` (`option_group_id`, {localize field='label'}`label`{/localize}, `value`, `name`, `weight`, {localize field='description'}`description`{/localize}, `is_active`) VALUES (@dashId, {localize}'{ts escape="sql"}Grants{/ts}'{/localize}, @maxValue, 'CiviGrant', @maxWeight, {localize}'Grants on dashboard'{/localize}, 1);
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '{localize field='label'}`label`{/localize}, `value`, `name`, `weight`, {localize' at line 1
MariaDB [civicrm]>
MariaDB [civicrm]> SELECT @statusId := id FROM `civicrm_option_group` WHERE `name` = 'grant_status';
+-----------------+
| @statusId := id |
+-----------------+
|              23 |
+-----------------+
1 row in set (0.00 sec)

MariaDB [civicrm]>
MariaDB [civicrm]> SELECT @maxValue := MAX( CAST( `value` AS UNSIGNED ) ) + 1 FROM  `civicrm_option_value` WHERE `option_group_id` = @statusId;
+-----------------------------------------------------+
| @maxValue := MAX( CAST( `value` AS UNSIGNED ) ) + 1 |
+-----------------------------------------------------+
|                                                   8 |
+-----------------------------------------------------+
1 row in set (0.00 sec)

MariaDB [civicrm]>
MariaDB [civicrm]> SELECT @maxWeight := MAX( CAST( `weight` AS UNSIGNED ) ) + 1 FROM  `civicrm_option_value` WHERE `option_group_id` = @statusId;
+-------------------------------------------------------+
| @maxWeight := MAX( CAST( `weight` AS UNSIGNED ) ) + 1 |
+-------------------------------------------------------+
|                                                     8 |
+-------------------------------------------------------+
1 row in set (0.00 sec)

MariaDB [civicrm]>
MariaDB [civicrm]> INSERT IGNORE INTO `civicrm_option_value` (`option_group_id`, {localize field='label'}`label`{/localize}, `value`, `name`, `weight`, `is_active`) VALUES (@statusId, {localize}'{ts escape="sql"}Draft{/ts}'{/localize}, @maxValue, 'Draft', @maxWeight, 1);
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '{localize field='label'}`label`{/localize}, `value`, `name`, `weight`, `is_activ' at line 1
MariaDB [civicrm]>
`

**Comment by JoeMurray** [comment]
Let's update this with a new release that addresses concerns above.

**Comment by lathem** [comment]
Just wanted to thank you guys for your rapid responses and actually caring about this issue.  I work with some big proprietary medical records stuff, and I _wish_ their (very expensive) tech support was even a quarter as good as you.

**Comment by Edzelopez** [comment]
@lathem you're welcome!

I am working on a new release and will update you when it is ready shortly.

**Comment by Edzelopez** [comment]
Hi @lathem,

I have added a [pre-release](https://github.com/JMAConsulting/biz.jmaconsulting.grantapplications/releases/tag/v1.5.2) for 4.7.22. We would appreciate it if you could let us know how it works for you and report any bugs if found.

Thanks!

**Comment by lathem** [comment]
Hi @Edzelopez and @JoeMurray --

Pardon taking a couple days to get back on this-- went ahead and dropped your pre-release into my 4.7.22 install and it seems to be working as expected.  

I did have a problem with a wordpress plugin (Justified Gallery) throwing a javascript namespace error problem when selecting profiles to add to the grant application page, but when I disabled that plugin, everything worked as expected.   I don't think it has anything to do with you guys-- I found this posting that described the same problem with a different plugin: https://civicrm.stackexchange.com/questions/3711/backbone-js-error-with-4-6-4-on-wp-messing-up-contribution-page-set-up

However, I think this is going to work.  I'll let you know if I experience any weirdness as I get this set up with the customer.  Thanks very much for the help!
